target_snowflake/db_sync.py

```diff
@@ -388,40 +411,55 @@ def put_to_stage(self, file, stream, count, temp_dir=None):
         self.logger.info('Uploading %d rows to stage', count)
         return self.upload_client.upload_file(file, stream, temp_dir)
 
-    def delete_from_stage(self, stream, s3_key):
+    def delete_from_stage(self, stream, upload_key):
         """Delete file from snowflake stage"""
-        self.logger.info('Deleting %s from stage', format(s3_key))
-        self.upload_client.delete_object(stream, s3_key)
+        self.logger.info('Deleting %s from stage', format(upload_key))
+        self.upload_client.delete_object(stream, upload_key)
 
-    def copy_to_archive(self, s3_source_key, s3_archive_key, s3_archive_metadata):
+    def copy_to_archive(self, upload_source_key, archive_key, archive_metadata):
         """
         Copy file from snowflake stage to archive.
 
-        s3_source_key: The s3 key to copy, assumed to exist in the bucket configured as 's3_bucket'
+        upload_source_key: The s3/azure key to copy, assumed to exist in the bucket configured as 's3_bucket' or 'azure_container'
 
-        s3_archive_key: The key to use in archive destination. This will be prefixed with the config value
+        archive_key: The key to use in archive destination. This will be prefixed with the config value
                         'archive_load_files_s3_prefix'. If none is specified, 'archive' will be used as the prefix.
 
                         As destination bucket, the config value 'archive_load_files_s3_bucket' will be used. If none is
                         specified, the bucket configured as 's3_bucket' will be used.
 
-        s3_archive_metadata: This dict will be merged with any metadata in the source file.
+        archive_metadata: This dict will be merged with any metadata in the source file.
 
         """
-        source_bucket = self.connection_config.get('s3_bucket')
-
-        # Get archive s3_bucket from config, or use same bucket if not specified
-        archive_bucket = self.connection_config.get('archive_load_files_s3_bucket', source_bucket)
-
         # Determine prefix to use in archive s3 bucket
         default_archive_prefix = 'archive'
-        archive_prefix = self.connection_config.get('archive_load_files_s3_prefix', default_archive_prefix)
-        prefixed_archive_key = f'{archive_prefix}/{s3_archive_key}'
 
-        copy_source = f'{source_bucket}/{s3_source_key}'
+        if self.staging == 'azure':
+            source_container = self.connection_config.get('azure_container')
+            # Get archive container from config, or use same container if not specified if not specified
+            archive_container = self.connection_config.get('archive_load_files_azure_container', source_container)
+            archive_prefix = self.connection_config.get('archive_load_files_azure_container_prefix', default_archive_prefix)
+
+            prefixed_archive_container = f'{archive_container}/{archive_prefix}'
```

Comment:
It might make sense to create another directory in this path. For example when doing a full refresh of `rpt-Salty` we create ~60 .csv files that are written to `archive`. The filepath could be something like `<archive_container>/<stream_name>_<stream_start_timestamp>/<archive_prefix>` (`archive/us-sensitive/Salty_2022-03-21...`). This way all of the files from one run would be archived together in one directory.